specs/destination_test.go

```diff
@@ -54,3 +40,137 @@ func TestDestinationUnmarshalSpec(t *testing.T) {
 		t.Fatalf("expected postgres://user:pass@host:port/db, got %s", spec.ConnectionString)
 	}
 }
+
+var destinationUnmarshalSpecTestCases = []struct {
+	name   string
+	spec   string
+	err    string
+	source *Source
+}{
+	{
+		"invalid_kind",
+		`kind: nice`,
+		"failed to decode spec: unknown kind nice",
+		nil,
```

Comment:
nit: let's add keys to these struct instantiations

Reply:
I feel for this test driven use-case it reduces the boilerplate imo and makes it easier to add new tests. just opinion ofc.

Reply:
Makes sense, though as someone reading it for the first time, I found it hard to know what all these parameters were for, and had to scroll up and down a few times. I think it improves readability if you have the keys right there